src/ape/api/providers.py

```diff
@@ -960,18 +966,15 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
             if "nonce too low" in str(vm_err):
                 # Add additional nonce information
                 new_err_msg = f"Nonce '{txn.nonce}' is too low"
-                raise VirtualMachineError(
+                vm_err = VirtualMachineError(
                     base_err=vm_err.base_err, message=new_err_msg, code=vm_err.code
-                ) from err
-
-            else:
-                raise vm_err from err
+                )
 
-        required_confirmations = (
-            txn.required_confirmations
-            if txn.required_confirmations is not None
-            else self.network.required_confirmations
-        )
+            vm_err.receipt = self.get_receipt(
```

Comment:
I want to make sure that if for any reason the receipt is non-fetchable, we can carry on.
Like I was going to suggest maybe putting it in a blank `Exception` `try/catch`.

Tests seem to be working though!
But I'd like to make sure we aren't futzing with anything on live network usages.

(task for anyone)
- [ ] See how reverts and other failures work on live networks with these changes

Reply:
Looking through the call chain for `get_receipt()` it seems it can raise `TransactionNotFoundError` or `TransactionError`

Updated in 84b6a53e but will leave this open because there will probably be some more changes around this